### ISSUE TITLE
Sensors for consumed heating & consumed water heating subtract energy usage every night

### ISSUE THREAD

**Issue by Schlichi83** [pail23/stiebel_eltron_isg_component#202]
I have a graph (apex chart) for my consumed energy for heating and water heating using. For about a week now the chart doesn't work anymore properly. 

I use the following sensors:
sensor.stiebel_eltron_isg_consumed_heating_today
sensor.stiebel_eltron_isg_consumed_water_heating_today

![Graph](https://github.com/pail23/stiebel_eltron_isg_component/assets/96698950/0e24c284-810a-479a-a54c-32aee5f8696f)

While the day is still running, everything is okay - but the next day the value for the previous day is reset to 0. I checked the statistics data and saw that every day at 23:55 the energy usage for the whole day is subtracted - see screenshow below.

![DailySensor](https://github.com/pail23/stiebel_eltron_isg_component/assets/96698950/d22e4866-8ec6-48d5-945f-a6717776a1f3)

If I correct the statistic value and set this to 0 everything works again. But this isn't a proper solution to manually correct this every day. Is this intentional? Because up to a week ago everything worked for almost a year. I know I could use the total sensors, but then I'd be missing today's values.

**Comment by Padanian** [comment]
The total of the day is reset around midnight. It seems that this is an issue pertaining to home assistant, more than to the integration. It should save the data around 23.45, instead than at midnight.

**Comment by Schlichi83** [comment]
Okay, if this is the case - how can I bring this up to the Home Assistant developers?

**Comment by Padanian** [comment]
will you try and see whether that would solve the issue?

**Comment by Padanian** [comment]
Yeah, needs to wait tomorrow. I browser the changelog of 2024.3 and there's nothing that could have affected the total.
I hope Paul releases soon the fix I prepared few days ago.

**Comment by pail23** [comment]
The xy_today energy sensors are not suitable for the energy dashboard since the counter resets every day to 0. I have added in the release 2024.2.2 new energy sensors (e.g. Produced heating). They should work fine in the energy dashboard. Please test.

**Comment by kucerazdenek** [comment]
Then we will loose the history in energy panel I am afraid?

**Comment by Padanian** [comment]
IMHO a monotonically increasing positive total cannot go back to zero. We should notify home assistant that the description is misleading. In fact, #153 error was caused by home assistant correctly catching a decreasing value for a monotonically increasing variable. If we revert #177 we will be back to square one with the same error.

**Comment by kucerazdenek** [comment]
@pail23 , @Padanian , just tested updated version of integration 2024.2.2 and new cumulative sensors. Unfortunately those don't work by design.
Sensor stiebel_eltron_isg_consumed_heating_today is reset to 0 around 11:55-11:58 pm (before midnight), while stiebel_eltron_isg_consumed_heating_total is updated after midnight by ISG. 
This leads to the subtraction of the daily consumption from the current day first (negative graph) and then to the addition of this consumption at the start of the following day (positive graph with consumption equal to previous day for first hour) as shown on pictures below.
I would intercede to revert #177 as proposed by @CSchlipp, because daily sensors worked fine till this change. Discussion about meaning of monotonically increasing positive is just formal, I believe that primary objective is to have working solution. Also, if you plan to remove daily consumption sensors as mentioned in release notes, energy graphs for more than year (in my case) will get useless then.

I have to mention that I really appreciate all the hard work you do in your spare time and for free, so that BFU like me can integrate a heat pump into HA and manage my household.

**New cumulated sensor stiebel_eltron_isg_consumed_heating**
![image](https://github.com/pail23/stiebel_eltron_isg_component/assets/121700009/75b7397b-ff59-4513-a5f1-ce10c7fdbbfa)

**End of previous day, new cumulative and original daily sensor in negative**
![image](https://github.com/pail23/stiebel_eltron_isg_component/assets/121700009/97923b29-cee2-453e-8da7-44f74dd6f707)

**Start of new day, cumulative sensor shows positive value with amount of energy consumed during previous day**
![image](https://github.com/pail23/stiebel_eltron_isg_component/assets/121700009/cce21093-63a6-4eeb-bf47-e29da5d71290)

**Comment by kucerazdenek** [comment]
OK, because there is no response since last week, I have changed the below in my sensor.py.
No subtraction of daily sensors at the end of the day anymore.
Please fix it in this matter in next release.
Thanks a lot.

`state_class=SensorStateClass.TOTAL,`

to

`state_class=SensorStateClass.TOTAL_INCREASING,`

```
def create_daily_energy_entity_description(name, key, visible_default=True):
    """Create an entry description for a energy sensor."""
    return SensorEntityDescription(
        key,
        name=name,
        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
        icon="mdi:meter-electric",
        has_entity_name=True,
        state_class=SensorStateClass.TOTAL_INCREASING,
        device_class=SensorDeviceClass.ENERGY,
        entity_registry_visible_default=visible_default,
    )

```

**Comment by starlight2k** [comment]
- the problem was already described somewhere... you can only read full kwh with modus
- isg webinterface shows three decimals

what ive seen:
the values are not live and only refresh hourly (?)... 

the shown kwh are 10-15% less than real consumption. 
i installed a shelly 3em for comparison

![Screenshot 2024-04-05 235227](https://github.com/pail23/stiebel_eltron_isg_component/assets/93672472/02351713-dc6f-4aca-8cf7-0c31683a3033)
![Screenshot 2024-04-05 235245](https://github.com/pail23/stiebel_eltron_isg_component/assets/93672472/ecef9dc8-385a-4268-9c76-ce3589fa8166)

**Comment by Padanian** [comment]
It is apparent that the storage of canbus values into modbus registers isn't working as expected.
I would suggest to refrain using hourly data and scrap it altogether and just use daily cumulative data.

**Comment by pail23** [comment]
I have reverted the changes to the xy_today and xy_total sensors. In addition, I have improved the spike detection logic for the following sensors:
- Produced Heating
- Produced Water Heating
- Consumed Heating
- Consumed Water Heating

Eventually these sensor might replace the xy_total and xy_today sensor.

Please provide me feedback if the sensors above are properly working for you.
